arch/arm/core/aarch32/cortex_a_r/reset.S

```diff
@@ -201,27 +204,57 @@ EL1_Reset_Handler:
      */
 
     /* FIQ mode stack */
+#if CONFIG_ASSEMBLER_ISA_THUMB2
```

Comment:
I wonder if we can remove the `#if CONFIG_ASSEMBLER_ISA_THUMB2` and just rewrite with
```
mov r0, #(MODE_FIQ | I_BIT | F_BIT)
msr CPSR_c, r0
```
regardless of THUMB2. There are really too many `#if` defines.

Also, I wonder what will happen if we are going to execute `it ne` in non-Thumb2 mode? Ignored or trap to instruction under exception? The reason I ask this question is just that I think there should be less #if define macros

Reply:
> I wonder if we can remove the #if CONFIG_ASSEMBLER_ISA_THUMB2 and just rewrite with ... regardless of THUMB2. There are really too many #if defines.

yes, it can be done. It will be perhaps redundant for A32 and somebody later may try to "optimize" it. But as long as we have some board in CI building in T32 we can prevent any situation like that.

>  I wonder what will happen if we are going to execute it ne in non-Thumb2 mode?

My understanding is that `IT` is a pseudo-instruction that does not generate any code in A32. From armasm https://developer.arm.com/documentation/dui0473/m/arm-and-thumb-instructions/it . With Zephyr SDK GCC, there's indeed no code generated for the `IT` when in A32. So we could also remove the guarding here.

Reply:
Sounds perfect, and if you worry about someone "optimize" it, then we could have comments telling everyone we need this because of THUMB2. What's your opinion?

Reply:
And we could wait for a while, in case there is a strong objection:)

Reply:
hi @microbuilder @stephanosio could please take a look to this patch and consider above suggestions? thanks

Reply:
> hi @microbuilder @stephanosio could please take a look to this patch and consider above suggestions? thanks 

Sorry for the slow reply. I'm on vacation this week, and this needs a bit of time to look at properly. 😬 I'll have a look at it as soon as I'm back in the office, though.

Reply:
> Sounds perfect, and if you worry about someone "optimize" it, then we could have comments telling everyone we need this because of THUMB2. What's your opinion?

I think just adding an appropriate comment is fine here for now, but it's better to leave a signpost about what's going on for future users.

Reply:
guards removed whenever possible